Lesson9 (Unordered containers)/main.cpp

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include <iterator>
+#include <unordered_map>
+#include <vector>
+#include <fstream>
+#include <experimental/filesystem>
+#include <algorithm>
+
+namespace fs = std::experimental::filesystem;
+
+struct compareTwoFiles{
+    bool operator()(const std::string& lhs, const std::string& rhs) const
+    {
+        std::ifstream lhsFile(lhs, std::ifstream::binary);
+        std::ifstream rhsFile(rhs, std::ifstream::binary);
+
+        if(lhsFile.is_open() == false  || rhsFile.is_open() == false)
+        {
+            return false;
+        }
+
+        return  std::equal(std::istream_iterator<std::string>(lhsFile),
+                           std::istream_iterator<std::string>(),
+                           std::istream_iterator<std::string>(rhsFile));
+    }
+};
+
+struct hashFunc{
+    size_t operator()(const std::string& key) const
+    {
+        size_t hash {0};
```

Comment:
**You didn't get the hash function application.
Please read https://honest.engineering/posts/hash-functions-survival-guide.
Not accepted. Rework it!**